Tests/Unit/EmogrifierTest.php

```diff
@@ -587,6 +587,10 @@ public function selectorDataProvider()
             => ['p:last-child {' . $styleRule . '} ', '#<p class="p-6" style="' . $styleRule . '">#'],
             'DIV:last-child does not match last child with mismatching tag'
             => ['div:last-child {' . $styleRule . '} ', '#<p class="p-6">#'],
+            'P:not(.p-6) matches all but P.p-6'
+            => ['p:not(.p-6) {' . $styleRule . '}', '#<p class="p-6">#'],
+            'P:not(.p-6) matches all but P.p-6 check if other paragraphs got style applied'
+            => ['p:not(.p-6) {' . $styleRule . '}', '#<p class="p-5" ' . $styleAttribute . '>#'],
```

Comment:
After #417 has been merged, you can also add this to matchedCssDataProvider and nonMatchedCssDataProvider, if you like. (You can also keep this as is, and then I'll convert this later.)

Reply:
I'll leave it to you then